classes/local/step/connector_sftp_delete_file.php

```
@@ -0,0 +1,129 @@
+class connector_sftp_delete_file extends connector_sftp {
+    use sftp_trait{
+        sftp_trait::form_define_fields as  sftp_form_define_fields;
+        sftp_trait::form_add_custom_inputs as sftp_form_add_custom_inputs;
+        sftp_trait::validate_for_run as sftp_validate_for_run;
+        sftp_trait::validate_config as sftp_validate_config;
+
+
+    }
+
+    /**
+     * Return the definition of the fields available in this form.
+     *
+     * @param string $behaviour 'delete' or something else.
+     * @return array
+     */
+    public static function form_define_fields($behaviour = 'delete'): array {
+        $fields = self::sftp_form_define_fields('delete');
+        return $fields;
+    }
+
+    /**
+     * Custom elements for editing the step.
+     *
+     * @param \MoodleQuickForm $mform
+     * @param string $behaviour default to the step.
+     */
+    public function form_add_custom_inputs(\MoodleQuickForm &$mform, $behaviour = 'delete') {
+        $this->sftp_form_add_custom_inputs($mform, 'delete');
+    }
+
+    /**
+     * Executes the step
+     *
+     * Performs an SFTP call according to config parameters.
+     *
+     * @param  mixed|null $input
+     * @return mixed
+     */
+    public function execute($input = null) {
+        $stepvars = $this->get_variables();
+        $config = $stepvars->get('config');
+
+        // At this point we need to disconnect once we are finished.
+        try {
+            // Skip if it is a dry run.
+            if ($this->is_dry_run() && $this->has_side_effect()) {
+                return $input;
+            }
+
+            $sftp = $this->init_sftp($config);
+
+            $sourceisremote = helper::path_is_scheme($config->source, self::$sftpprefix);
+            $sourcepath = $this->resolve_path($config->source);
+
+            // Deleting from remote.
+            if ($sourceisremote) {
+                $this->delete_from_remote($sftp, $sourcepath);
+                return $input;
+            }
```

If the source isn't a remote, it should throw an exception here. Because it's validated at form validation time, if this is not a remote here it means something is very wrong.

---

An exception has been raised here